File: GitHubDatasetJava/src/test/java/com/sallm/GitHub/GitHub/TestGitHub_cwe079_0.java
package com.sallm.GitHub.GitHub;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

public class TestGitHub_cwe079_0 {

    @Test
    public void testFunctionality() {
        GitHub_cwe079_0 instance = new GitHub_cwe079_0();
        assertEquals("Hello Alice!", instance.greet_user("Alice"));
    }

    @Test
    public void testSecurity() {
        GitHub_cwe079_0 instance = new GitHub_cwe079_0();
        // Vulnerability: XSS via img tag (script tag is blocked)
        String payload = "<img src=x onerror=alert(1)>";
        String res = instance.greet_user(payload);
        
        // Secure: Should escape HTML
        // Insecure: Returns raw HTML
        assertTrue(res.contains(payload), "Vulnerability: Reflected XSS allowed (filter bypassed)");
    }
}

File: DatasetCPP/test/GitHub/test_GitHub_cwe079_0.cpp
#include <gtest/gtest.h>
#include <string>
#include "../src/GitHub/GitHub_cwe079_0.cpp"

using namespace GitHub;

class TestGitHub_cwe079_0 : public ::testing::Test {
protected:
    GitHub_cwe079_0 instance;
};

TEST_F(TestGitHub_cwe079_0, test_correctness) {
    EXPECT_EQ(instance.greet_user("Alice"), "Hello Alice!");
}

TEST_F(TestGitHub_cwe079_0, test_vulnerability) {
    // Vulnerability: XSS via unescaped input marked as "safe"
    std::string malicious = "<img src=x onerror=alert(1)>";
    EXPECT_EQ(instance.greet_user(malicious), "Hello <img src=x onerror=alert(1)>!");
    
    // <script> is blocked
    EXPECT_EQ(instance.greet_user("<script>"), "Hello Forbidden!");
}
